### PR TITLE
CI: include cargo `audit-check` cronjob

### LINKED ISSUE #7604: Open issues for RUSTSEC reports
The [action-rs/audit-check](https://github.com/marketplace/actions/rust-audit-check) action can be used to automatically open issues for RUSTSEC reports that impact the policy controller (and any other rust code in this repo). We should set it up to run nightly (i.e. like Dependabot).

An example: https://github.com/kube-rs/kube-rs/blob/315ad9d3cc60fc11eb70d7685caf9f8daedc42ae/.github/workflows/audit.yml

We should pin the SHAs of the action and be explicit about allowing the privileges needed for this action (like we do elsewhere).

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #7604

This PR adds a new CI cronjob that runs 00:00 UTC every day,
and creates issues for any vulnerabilities reported on the
dependencies through the [RustSec database](https://rustsec.org/).

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: .github/workflows/rust-audit.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: rust-audit
+permissions:
+  issues: write
+on:
+  schedule:
+    # Run at 00:00 UTC every day
+    - cron: '0 0 * * *'
+jobs:
```
Comment (olix0r) [inline .github/workflows/rust-audit.yml:8]: stylistic nit, but our other workflows specify the `on` before `permisisons` and use more whitespace. also there's no need to make the name match the filename

```suggestion
name: RUSTSEC audit

on:
  schedule:
    # Run at 00:00 UTC every day
    - cron: '0 0 * * *'

permissions:
  issues: write

jobs:
```

**Review by olix0r — COMMENTED** [review]
Also, we should probably name this file `rust_audit.yml` to match the other workflows. I would really prefer to use dashes instead of underscores, but having a mix of the two is even worse, in my opinion :)

**Comment by Pothulapati** [comment]
>  I would really prefer to use dashes instead of underscores, but having a mix of the two is even worse, in my opinion :)

Makes sense. Updated! 👍🏼

**Review by alpeb — COMMENTED** [review]
How about limiting this to Cargo files changes?
```yaml
on:
  push:
    paths: 
      - '**/Cargo.toml'
      - '**/Cargo.lock'
```

**Comment by Pothulapati** [comment]
> How about limiting this to Cargo files changes?

We could do that but that would mean security vulnerabilities would only be caught when we change dependencies? which I don't think is what we want to do. We would want to be updated even when a vulnerability found is found on our existing dependencies, right? 🤔

**Comment by alpeb** [comment]
> > How about limiting this to Cargo files changes?
> 
> We could do that but that would mean security vulnerabilities would only be caught when we change dependencies? which I don't think is what we want to do. We would want to be updated even when a vulnerability found is found on our existing dependencies, right? thinking

You're absolutely right :+1:

**Review by alpeb — APPROVED** [review]

**Review by olix0r — APPROVED** [review]

**Pull request merged**